In [3]:
import cv2 as cv
import numpy as np

## Reading image

In [3]:
image = cv.imread('Photos/cat.jpg')
cv.imshow('Cat', image)
cv.waitKey(0)

-1

## Reading videos

In [5]:
 # 0 for webcam, 1 for first usb cam and so on
capture = cv.VideoCapture('Videos/dog.mp4')
while True:
    isTrue, frame = capture.read()
    cv.imshow('Video', frame)
    if cv.waitKey(20) & 0xFF == ord('d'):
        break
        
capture.release()
cv.destroyAllWindows()

error: OpenCV(4.5.3) C:\Users\runneradmin\AppData\Local\Temp\pip-req-build-sn_xpupm\opencv\modules\imgproc\src\color.cpp:182: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'


We get 215 assertion failed error when opencv could not find a media file at that particular location that you specified. Now we got the error because the video ran out of frames

## Rescale image

In [3]:
# This works for videos, images, live video
def rescaleFrame(frame, scale = 0.75):
    width = int(frame.shape[1] * scale)
    height = int(frame.shape[0] * scale)
    dimensions = (width, height) 
    return cv.resize(frame, dimensions, interpolation = cv.INTER_AREA)

In [4]:
capture = cv.VideoCapture('Videos/dog.mp4')
while True:
    isTrue, frame = capture.read()
    frameResized = rescaleFrame(frame, 0.5)
    cv.imshow('Rescaled Video', frameResized)
    cv.imshow('Video', frame)
    if cv.waitKey(20) & 0xFF == ord('d'):
        break
        
capture.release()
cv.destroyAllWindows()

In [5]:
# Only works for live video
def changeResolution(width, height):
    capture.set(3, width)
    capture.set(4, height)

## Draw

In [4]:
# creating a blank image
blank = np.zeros((500, 500, 3), dtype = 'uint8')

### Print the image a certain colour

In [5]:
blank[:] = 0,255,0
cv.imshow('Green', blank)